lib/transport.js

```diff
@@ -0,0 +1,80 @@
+'use strict'
+
+const { createRequire } = require('module')
+const caller = require('get-caller-file')
+const { join, isAbsolute } = require('path')
+
+const ThreadStream = require('thread-stream')
+
+function buildStream (filename, workerData, workerOpts) {
+  const stream = new ThreadStream({
+    filename,
+    workerData,
+    workerOpts,
+    sync: true // TODO should this be configurable?
+  })
+
+  stream.on('ready', function () {
+    stream.unref()
+
+    if (workerOpts.autoEnd !== false) {
+      // TODO possibly use FinalizationGroup to automatically remove
```

Comment:
TODO?

Reply:
I prefer to not resolve this here. There is a space for a tiny little utility for doing exactly this, however it's better developed in isolation. It's more a note for improvements in the future.

This means we are leaking memory if a transport is not closed. I think it's ok.